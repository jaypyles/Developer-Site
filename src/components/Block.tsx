import React from "react";

type BlockProps = {
  link: string;
  small_description: string;
};

const Block: React.FC<BlockProps> = ({ link, small_description }) => {
  return (
    <div className="block flex flex-col bg-testAccent rounded mb-[1em]">
      <a href={link} className="link text-[75%] !ml-[0.5em]">
        {link}
      </a>
      <p className="description text-[60%] ml-[0.5em] mb-[0.5em]">
        {small_description}
      </p>
    </div>
  );
};

export default Block;
