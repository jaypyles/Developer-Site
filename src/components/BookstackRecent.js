import React, { useState, useEffect } from "react";
import Constants from "../constants";

const BookstackRecent = () => {
  const [pageURL, setPageURL] = useState("");
  const url = `${Constants.DOMAIN}/api/bookstack/recent-page`;

  const fetchBookstack = async () => {
    try {
      const res = await fetch(url);
      if (res.ok) {
        const data = await res.json();
        setPageURL(JSON.parse(data));
      } else {
        console.error("Failed to fetch from API: ", res.status);
      }
    } catch (error) {
      console.error("Failed to fetch from API: ", error);
    }
  };

  useEffect(() => {
    fetchBookstack();
  }, []);

  return (
    <div className="bookstack-recent">
      {pageURL ? (
        <p>
          Most recent note: <a href={pageURL.url}>{pageURL.url}</a>
        </p>
      ) : (
        <></>
      )}
    </div>
  );
};

export default BookstackRecent;
