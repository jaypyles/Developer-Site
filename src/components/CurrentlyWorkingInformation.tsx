import React from "react";

interface Data {
  image: string;
  title: string;
  description: string;
}

interface CurrentlyWorkingInformationProps {
  data: Data;
}

const CurrentlyWorkingInformation: React.FC<
  CurrentlyWorkingInformationProps
> = ({ data }) => {
  const imageUrl = `/images/${data.image}`;

  return (
    <div className="workblock flex flex-row mr-[0.5em] p-1 bg-imageAccent emboss">
      {imageUrl && (
        <img
          className="image w-[4em] h-[4em] mr-[0.5em]"
          src={imageUrl}
          alt={data.title}
        />
      )}
      <div className="content mt-[0.25]">
        <p className="title text-[75%] mb-0">{data.title}</p>
        <p className="description text-[65%] mb-0 text-gray-800">
          {data.description}
        </p>
      </div>
    </div>
  );
};

export default CurrentlyWorkingInformation;
