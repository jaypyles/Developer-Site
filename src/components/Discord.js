import React, { useState, useEffect } from "react";
import { styled } from "@mui/material/styles";
import { Grid, Avatar, Badge, Typography } from "@mui/material";
import StyledBadge from "./StyledBadge";

const Discord = () => {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(true);
  const domain = process.env.REACT_APP_DOMAIN;
  const url = `${domain}/api/discord/status`;

  useEffect(() => {
    fetch(url)
      .then((res) => res.json())
      .then((res) => {
        setData(res.data);
        setLoading(false);
      });
  }, []);

  return (
    <div className="discord">
      {!loading && data && (
        <div className="status">
          <div className="title">
            <p>Discord Status</p>
          </div>
          <div className="content">
            <div className="avatar">
              <StyledBadge
                overlap="circular"
                variant="dot"
                anchorOrigin={{
                  vertical: "bottom",
                  horizontal: "right",
                }}
                color={
                  data.discord_status === "online"
                    ? "success"
                    : data.discord_status === "dnd"
                      ? "error"
                      : data.discord_status === "idle"
                        ? "warning"
                        : "default"
                }
              >
                <Avatar
                  src={`https://cdn.discordapp.com/avatars/${data.discord_user.id}/${data.discord_user.avatar}.png`}
                />
              </StyledBadge>
            </div>
            <div className="user">
              <Typography variant="h6">
                {data.discord_user.username}
                <Typography
                  variant="subtitle1"
                  component="span"
                  sx={{ color: "gray" }}
                >
                  #1337
                </Typography>
              </Typography>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

export default Discord;
