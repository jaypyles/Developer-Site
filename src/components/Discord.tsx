import React, { useState, useEffect } from "react";
import { Avatar, Typography } from "@mui/material";
import StyledBadge from "./StyledBadge";
import { getDiscordStatus } from "src/lib/UtilFunctions";
import Image from "next/image";

interface Status {
  state: string;
  emoji: {
    id: string;
  };
}

interface DiscordData {
  discord_status: string;
  discord_user: {
    id: string;
    avatar: string;
    username: string;
    display_name: string;
  };
  activities: Status[];
}

interface DiscordProps {
  loadedState: {
    loaded: boolean;
    setLoaded: React.Dispatch<React.SetStateAction<boolean>>;
  };
}

const Discord: React.FC<DiscordProps> = ({ loadedState }) => {
  const [data, setData] = useState<DiscordData | null>(null);
  const { loaded, setLoaded } = loadedState;

  useEffect(() => {
    const fetchDiscord = async () => {
      try {
        const status = await getDiscordStatus();
        setData(status);
        setLoaded(true);
      } catch (error) {
        console.error("Error fetching Discord status:", error);
      }
    };

    fetchDiscord();
  }, []);

  return (
    <div className="discord">
      {loaded && data && (
        <div className="discord-status">
          <div className="content">
            <div className="avatar">
              <StyledBadge
                overlap="circular"
                variant="dot"
                anchorOrigin={{
                  vertical: "bottom",
                  horizontal: "right",
                }}
                color={
                  data.discord_status === "online"
                    ? "success"
                    : data.discord_status === "dnd"
                    ? "error"
                    : data.discord_status === "idle"
                    ? "warning"
                    : "default"
                }
              >
                <Avatar
                  sx={{ height: "100%", width: "100%" }}
                  src={`https://cdn.discordapp.com/avatars/${data.discord_user.id}/${data.discord_user.avatar}.png`}
                />
              </StyledBadge>
            </div>
            <div className="user flex flex-col justify-start w-1/2">
              <Typography variant="h6" style={{ fontWeight: "bold" }}>
                {data.discord_user.display_name}
              </Typography>
              <div className="flex flex-row align-middle">
                {data.activities[0]!.emoji && (
                  <Image
                    className="mr-[0.15rem] h-fit"
                    alt="Discord Status Emoji"
                    height={16}
                    width={12}
                    src={`https://cdn.discordapp.com/emojis/${
                      data.activities[0]!.emoji.id
                    }`}
                  ></Image>
                )}
                <Typography
                  variant="subtitle1"
                  component="span"
                  sx={{ color: "gray", fontSize: "0.75em", lineHeight: 1 }}
                >
                  {data.activities[0]!.state}
                </Typography>
              </div>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

export default Discord;
