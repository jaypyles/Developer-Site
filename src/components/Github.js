import React, { useState, useEffect } from "react";

const Github = () => {
  const [githubURL, setGithubURL] = useState("");
  const domain = process.env.REACT_APP_DOMAIN;
  const url = `${domain}/api/github/recent`;

  const fetchGithub = async () => {
    try {
      const res = await fetch(url);
      if (res.ok) {
        const data = await res.json();
        setGithubURL(JSON.parse(data));
      } else {
        console.error("Failed to fetch Github data:", res.status);
      }
    } catch (error) {
      console.error("Failed to fetch Github data:", error);
    }
  };

  useEffect(() => {
    // fetch most recent on mount
    fetchGithub();
  }, []);

  return (
    <div className="github text-center">
      {githubURL ? (
        <>
          <p>
            Most recent project: <a href={githubURL.url}>{githubURL.url}</a>
          </p>
        </>
      ) : (
        <></>
      )}
    </div>
  );
};

export default Github;
