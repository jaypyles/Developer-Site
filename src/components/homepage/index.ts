export * from "./currently-working";
export * from "./homepage";
export * from "./current-theme";
