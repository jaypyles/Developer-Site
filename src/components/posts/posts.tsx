import {
  Box,
  ImageList,
  ImageListItem,
  ImageListItemBar,
  Modal,
  Paper,
} from "@mui/material";
import clsx from "clsx";
import { useState } from "react";
import { CardPost } from "src/components/posts";
import classes from "./posts.module.css";

interface PostDocument {
  image_id: string;
  description: string;
  time_posted: string;
}

interface PostsProps {
  posts: PostDocument[];
}

export const Posts: React.FC<PostsProps> = ({ posts }) => {
  const [open, setOpen] = useState(false);
  const [photo, setPhoto] = useState({
    image_id: "",
    description: "",
    time_posted: "",
  });

  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);

  const style = {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 500,
    bgcolor: "#c3c6cb",
    boxShadow: 24,
    p: 4,
  };

  return (
    <div id="post-wrapper">
      {posts && posts.length > 0 ? (
        <div id="posts" className="emboss-no-top !font-prompt">
          <Paper
            id="post-array"
            className="p-2 no-scrollbar"
            sx={{ bgcolor: "#c3c6cb" }}
          >
            <ImageList cols={3} gap={8} className="post-list">
              {posts.map((item, index) => (
                <ImageListItem key={index}>
                  <img
                    src={`/api/post_images/${item.image_id}?w=248&fit=crop&auto=format&dpr=2&compress=true`}
                    alt={item.description}
                    id="post-img"
                    className={clsx(
                      "max-w-[15vw] cursor-pointer transition ease-in-out delay-50 hover:brightness-75 duration-100 emboss",
                      classes.postImg
                    )}
                    onClick={() => {
                      setPhoto(item);
                      handleOpen();
                    }}
                  />
                  <ImageListItemBar
                    className="!text-zinc-700"
                    subtitle={item.time_posted}
                  />
                </ImageListItem>
              ))}
            </ImageList>
          </Paper>

          <Modal
            open={open}
            onClose={handleClose}
            aria-labelledby="modal-modal-title"
            aria-describedby="modal-modal-description"
          >
            <Box sx={style} id="modal-box">
              <CardPost
                img={`/api/post_images/${photo.image_id}`}
                description={photo.description}
                date={photo.time_posted}
              />
            </Box>
          </Modal>
        </div>
      ) : (
        <></>
      )}
    </div>
  );
};
