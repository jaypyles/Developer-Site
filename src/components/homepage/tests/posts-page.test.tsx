import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { Posts } from "src/components/posts";

test("Render posts page properly.", () => {
  const result = render(<Posts posts={[]} />);
  const headingElement = result.container.querySelector("#post-wrapper");
  expect(headingElement).toMatchSnapshot();
});
