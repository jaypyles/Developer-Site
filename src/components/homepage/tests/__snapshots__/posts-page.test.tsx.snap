// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render posts page properly. 1`] = `
<div
  id="post-wrapper"
/>
`;
